### PR TITLE
NVMeFC NQN Changes

### PR DESCRIPTION
**Pull request by harshitap26**
# PR Submission checklist

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/260|

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [x] Did you run tests in a real Kubernetes cluster?
- [x] Have you maintained backward compatibility

## Description of your changes:
Added a new parameter to the REST API call api/rest/cluster for the adding the NVMe subsystem NQN for NVMe devices. Since the value is added only on Powerstore 3.0, check has been provided for the same. 

A single function GetSoftwareMajorVersion was created to remove the duplicity in the code.

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -15,6 +16,15 @@ func (c *ClientIMPL) GetCluster(ctx context.Context) (resp Cluster, err error) {
 	var systemList []Cluster
 	cluster := Cluster{}
 	qp := c.APIClient().QueryParamsWithFields(&cluster)
+
+	majorVersion, err := c.GetSoftwareMajorVersion(ctx)
+	if err != nil {
+		log.Printf("ERROR: Couldn't find the major array version")
```
Comment (harishp8889) [inline cluster.go:22]: Please change it info, error or debug statements, Printf logs may not appear when we set `X_CSI_DEBUG` to true

File: cluster_types.go [hunk 2]
```diff
@@ -30,6 +30,8 @@ type Cluster struct {
 	ManagementAddress string `json:"management_address,omitempty"`
 	// Current state of the cluster
 	State string `json:"state,omitempty"`
+	// NVMe Subsystem NQN for cluster
```
Comment (harishp8889) [inline cluster_types.go:33]: The comment should be  `NVMeNQN` right?

File: software_installed.go [hunk 3]
```diff
@@ -54,3 +56,28 @@ func (c *ClientIMPL) GetSoftwareInstalled(
...
+			if software.IsCluster == true {
+				softwareVersion = software.BuildVersion
+			}
+		}
+	}
+
+	if len(softwareVersion) > 0 {
+		majorVersion, err = strconv.Atoi(softwareVersion[0:1])
+		if err != nil {
+			log.Printf("ERROR: Couldn't convert the software version")
```
Comment (harishp8889) [inline software_installed.go:78]: same here, Change log.Printf to the respective log level type.

**Review by harishp8889 — CHANGES_REQUESTED** [review]

File: cluster_types.go (hunk 2 above)
Comment (harshitap26) [reply cluster_types.go:33]: Just added the usability similar to what is mentioned on the swagger, similar to the way mentioned for other parameters.

**Review by harishp8889 — APPROVED** [review]

File: cluster.go (hunk 1 above)
Comment (harshitap26) [reply cluster.go]: Updated with log.Errorf statements

File: software_installed.go (hunk 3 above)
Comment (harshitap26) [reply software_installed.go]: Updated with log.Errorf statements

**Pull request merged**